packages/transaction-builder/src/index.ts

```diff
@@ -1 +1,204 @@
-export {}
+import { Buffer } from 'buffer'
+import {
+  MsgProtoAppStake,
+  MsgProtoAppUnstake,
+  MsgProtoNodeStakeTx,
+  MsgProtoNodeUnjail,
+  MsgProtoNodeUnstake,
+  MsgProtoSend,
+} from './models/msgs'
+import { AbstractProvider } from '@pokt-foundation/pocketjs-abstract-provider'
+import { KeyManager } from '@pokt-foundation/pocketjs-signer'
+import {
+  RawTxRequest,
+  TransactionResponse,
+} from '@pokt-foundation/pocketjs-types'
+import { TxEncoderFactory } from './factory/tx-encoder-factory'
+import { TxMsg, CoinDenom, TxSignature } from './models/'
+
+export type ChainID = 'mainnet' | 'testnet'
+
+export const BASE_FEE = '10000'
+
+export class TransactionBuilder {
+  private provider: AbstractProvider
+  private signer: KeyManager
+  private chainID: ChainID
+
+  constructor({
+    provider,
+    signer,
+    chainID = 'mainnet',
+  }: {
+    provider: AbstractProvider
+    signer: KeyManager
+    chainID: ChainID
+  }) {
+    this.provider = provider
+    this.signer = signer
+    this.chainID = chainID
+  }
+
+  /**
+   * Signs and creates a transaction object that can be submitted to the network given the parameters and called upon Msgs.
+   * Will empty the msg list after succesful creation
+   * @param {string} fee - The amount to pay as a fee for executing this transaction
+   * @param {CoinDenom | undefined} feeDenom - The denomination of the fee amount
+   * @param {string | undefined} memo - The memo field for this account
+   * @returns {Promise<RawTxResponse | RpcError>} - A Raw transaction Response object or Rpc error.
+   * @memberof TransactionSender
+   */
+  public async createTransaction({
+    fee,
+    feeDenom = CoinDenom.Upokt,
+    memo = '',
+    txMsg,
+  }: {
+    fee: string | bigint
+    feeDenom: CoinDenom
+    memo: string
+    txMsg: TxMsg
+  }): Promise<RawTxRequest> {
+    // First, let's check if all required parameters are filled correctly
+    // Let's avoid a footgun: if coindenom is not UPokt, let's make sure fee is not
+    // BASE_FEE
+    if (feeDenom != CoinDenom.Upokt && fee == BASE_FEE) {
```

Comment:
```suggestion
    if (feeDenom != CoinDenom.Upokt && fee >= BASE_FEE_POKT) {
```

Reply:
With the current logic, someone could pay 9999 POKT or 10001 POKT fee which is as bad as 10000 [💀](https://emojipedia.org/skull/)

Reply:
Also, just noticed that `BASE_FEE` was a string. So we either need to cast it to number here or during const declaration

Reply:
See above.

As for the first commment, yeah for sure they could pay `9999` POKT haha, but they would have to intentionally write this. We wanna avoid paying a "default" fee in a way. Tbh I'm tempted to NOT give the option to pay in `POKT` and just hardcode it to `uPOKT`.

Reply:
> 

Sounds good tbh, not doing that can only go wrong

Reply:
In case you are still giving the option to pay fee in POKT, then:
```suggestion
    if (feeDenom != CoinDenom.Upokt && fee > 1) {
```

Reply:
Done in https://github.com/pokt-foundation/pocket-js-slim/pull/39/commits/2bbcaea7287c4382c05a0c12eb43a93f4b4e788f.